lamindb/models/artifact.py

```diff
@@ -1920,12 +1934,13 @@ def from_tiledbsoma(
 
             artifact = ln.Artifact.from_tiledbsoma("s3://mybucket/store.tiledbsoma", description="a tiledbsoma store").save()
         """
-        if UPath(path).suffix != ".tiledbsoma":
+        if not data_is_soma_experiment(exp):
             raise ValueError(
-                "A tiledbsoma store should have .tiledbsoma suffix to be registered."
+                "data has to be a SOMA Experiment object or a path to SOMA Experiment store."
             )
+        exp = exp.uri.removeprefix("file://") if not isinstance(exp, UPathStr) else exp
```

Comment:
What is this? Why is it needed?

Reply:
To get the local path, we need to remove this prefix which is always present in `experiment.uri`